File: Ejemplo_en_C/Programa_Video_Juego/Programa_Video_Juego/Clases/Enemigo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Programa_Video_Juego.Clases
{
    class Enemigo
    {
        private int vida;
        private Machete arma;

        public Enemigo(int vida)
        {
            this.vida = vida;
            arma = new Machete();
        }

        public int Vida { get => vida; set => vida = value; }

        public string Lanzar_Machete(Jugador objetivo)
        {
            objetivo.Vida -= arma.Daño;
            return "Vida del Soldado: " + objetivo.Vida;
        }
    }
}

File: Ejemplo en C/Clases/Enemigo.c
#include <stdlib.h>
#include <stdio.h>

#include "Enemigo.h"

Enemigo *Crear_Enemigo(int vida)
{
    Enemigo *nuevo_enemigo = (Enemigo*) malloc(sizeof(Enemigo));
    
    nuevo_enemigo->vida = vida;   
    Machete *nuevo_machete = Crear_Machete(10);
    nuevo_enemigo->arma = *nuevo_machete;  
}

void Destruir_Enemigo(Enemigo *este_enemigo, Machete *este_machete)
{
    free(este_enemigo);
    Destruir_Machete(este_machete);
}

void Lanzar_Machete(Jugador *objetivo){
    objetivo->vida = -10;
}